src/applications/piezoscan.py

```diff
@@ -98,6 +99,12 @@ def update(self, model):
                                                                    model.xmax + model.step_size,
                                                                    model.current_y + model.step_size,
                                                                    model.ymin])
+        
```

Comment:
Do we need / want to have the ability to mark a position on the scan "during" the scan?  When I click during a scan (in random data test mode on local Mac) I get sometime get errors on the console.

```
Traceback (most recent call last):
  File "/Users/adam/miniconda3/envs/qt3utilsdevpy312/lib/python3.12/site-packages/matplotlib/cbook.py", line 298, in process
    func(*args, **kwargs)
  File "/Users/adam/miniconda3/envs/qt3utilsdevpy312/lib/python3.12/site-packages/matplotlib/backend_bases.py", line 2972, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/adam/miniconda3/envs/qt3utilsdevpy312/lib/python3.12/site-packages/matplotlib/backend_bases.py", line 2957, in _mouse_event_to_message
    artists = [a for a in event.inaxes._mouseover_set
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: OrderedDict mutated during iteration
```

I think this is probably a race-condition. The `update` function is being called by two different threads: the main thread and the "scan thread". 

This might be fixed if we get rid of the for-loop around self.ax.lines.

Reply:
I have never gotten this error when clicking during a scan. I think the "feature" is somewhat nice however as I can look at positions and see the counts while the scan is running without having to wait for the scan to finish.